### PR TITLE
OCPBUGS-13978: bump k8s modules to 0.27.4

### PR DESCRIPTION
**Pull request by alebedev87**
Bump of `k8s.io/api@0.27.4`. `controller-runtime@0.15.1` bump was mandatory as it's the first version supporting k8s api >=0.27.x.

Changes related to `controller-runtime@0.15.1`:
- Manager options changed:
  - `NewClient` changed the signature and the cache handling
  - `Namespace` and `MultiNamespaceCacheBuilder` are deprecated now, `Cache` field was used instead
- `WithStatusSubresource` method has to be used in the fake client if the status is expected to be updated
- `source.Kind` is not public anymore
- `sigs.k8s.io/controller-runtime/pkg/envtest/printer` package has been removed
- `webhook.Validator` interface changed the signature of its methods

Also `golangci-lint` (part of `verify` job) has been updated with the `staticcheck` and `gosimple` linters which are [suggested to be default](https://golangci-lint.run/usage/linters/#enabled-by-default). The staticcheck found some deprecated methods:
- `https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait` deprecated a lot of `PollX` methods in favor of the context aware alternatives.
- `k8s.io/utils/pointer` deprecated all the methods with `Ptr` in the name in favor of the methods without it

### CODE REVIEW

**Comment by gcs278** [comment]
Reviewed and have no issues.
/lgtm

**Comment by gcs278** [comment]
/lgtm cancel
actually, I have one comment...sorry.

File: go.mod [hunk 1]
```diff
@@ -7,21 +7,21 @@ require (
...
 	github.com/openshift/api v0.0.0-20220906163444-2df055c101a3
 	github.com/openshift/cloud-credential-operator v0.0.0-20211118210017-9066dcc747fa
 	github.com/operator-framework/api v0.11.0
 	google.golang.org/api v0.58.0
-	k8s.io/api v0.25.3
-	k8s.io/apimachinery v0.25.3
-	k8s.io/client-go v0.25.3
-	k8s.io/utils v0.0.0-20220728103510-ee6ede2d64ed
-	sigs.k8s.io/controller-runtime v0.13.0
+	k8s.io/api v0.27.2
```
[the remark this answers is not in the document]
Comment (alebedev87) [reply go.mod]: Meantime they released `0.27.4`, so I bumped to it. Hope they'll not release any new before we decide to merge the PR :)

File: go.mod [hunk 2]
```diff
@@ -7,21 +7,21 @@ require (
...
 	github.com/operator-framework/api v0.11.0
 	google.golang.org/api v0.58.0
-	k8s.io/api v0.25.3
-	k8s.io/apimachinery v0.25.3
-	k8s.io/client-go v0.25.3
-	k8s.io/utils v0.0.0-20220728103510-ee6ede2d64ed
-	sigs.k8s.io/controller-runtime v0.13.0
+	k8s.io/api v0.27.4
+	k8s.io/apimachinery v0.27.4
+	k8s.io/client-go v0.27.2
```
Comment (gcs278) [inline go.mod:22]: Should this be bumped to 0.27.4 too?

Comment (alebedev87) [reply go.mod]: I think so, weird it wasn't bumped as a dependency.

File: go.mod [hunk 3]
```diff
@@ -7,21 +7,21 @@ require (
...
-	k8s.io/api v0.25.3
-	k8s.io/apimachinery v0.25.3
-	k8s.io/client-go v0.25.3
-	k8s.io/utils v0.0.0-20220728103510-ee6ede2d64ed
-	sigs.k8s.io/controller-runtime v0.13.0
+	k8s.io/api v0.27.4
+	k8s.io/apimachinery v0.27.4
+	k8s.io/client-go v0.27.4
+	k8s.io/utils v0.0.0-20230209194617-a36077c30491
+	sigs.k8s.io/controller-runtime v0.15.0
```
Comment (gcs278) [inline go.mod:24]: I feel bad for commenting this...https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.1

Thnk we should bump?

Comment (alebedev87) [reply go.mod:24]: Seems to be a bug fix release, let me try.

**Comment by gcs278** [comment]
Sorry for the delays.
/lgtm

**Pull request merged**